### PR TITLE
named entity recognition

### PR DESCRIPTION
**Pull request by xehu**
see: https://github.com/Watts-Lab/team-process-map/wiki/Named-Entity-Recognition

### CODE REVIEW

File: feature_engine/feature_builder.py [hunk 1]
```diff
@@ -306,6 +314,8 @@ def save_features(self) -> None:
         """
         # TODO: For now this function is very trivial. We will finalize the output formats (with date-time info etc) 
         # and control the output mechanism through this function.
+        # print(self.chat_data)
+        # self.chat_data.to_csv("C:/Users/amyta/Documents/GitHub/team-process-map/feature_engine/output/chat/jury_output_chat_level.csv", index=False)
```
[the remark this answers is not in the document]
Comment (amytangzheng) [reply feature_engine/feature_builder.py:318]: removed!

File: feature_engine/utils/calculate_chat_level_features.py [hunk 2]
```diff
@@ -101,6 +114,9 @@ def calculate_chat_level_features(self) -> pd.DataFrame:
         # Forward Flow
         self.get_forward_flow()
         self.get_certainty_score()
+        # # Forward Flow
```
[the remark this answers is not in the document]
Comment (amytangzheng) [reply feature_engine/utils/calculate_chat_level_features.py:117]: I think I had a formatting issue? but fixed!

File: feature_engine/featurize.py [hunk 3]
```diff
@@ -57,6 +82,7 @@
 	# testing chat features
 	testing_chat = FeatureBuilder(
 		input_df = chat_df,
+		ner_training_df = test_ner_training_df,
```
[the remark this answers is not in the document]
Comment (amytangzheng) [reply feature_engine/featurize.py]: Only test_ner_df has a training file. Just removed that line to test if it skips the feature if no dataframe is passed in

**Comment by xehu** [comment]
@amytangzheng Now that we have set up the public-facing documentation infrastructure, could you also add that in for the Named Entity Recognition feature before we merge it in?

Instructions are here: https://docs.google.com/document/d/1W7SHJTjqpOYreRaQERwi5xOOaLpYHxj_rbv02XazAWU/edit#heading=h.37xqjdsgdofj

**Pull request merged**